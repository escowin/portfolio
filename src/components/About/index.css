#about {
    display: grid;
    grid-template-columns: auto 1fr;
    gap: .5rem;
}

#profile-pic {
    border-radius: 50%;
    width: 45vw;
    margin: .5rem;
    border: 1px solid var(--dark-accent);
}

#profile-text h2 {
    margin: 2.5rem 0 .25rem 0;
    padding-right: .25rem;
    border-bottom: 1px solid var(--dark-accent);
    width: fit-content;
}

#bio {
    grid-column: span 2;
    display: flex;
    flex-direction: column;
    gap: 1rem;
    padding: 0 .75rem;
    font-family: 'Raleway', sans-serif;
}