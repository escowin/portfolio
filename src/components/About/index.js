import React from 'react';
import profilePic from '../../assets/images/profile.jpg'

function About() {
    return (
      <section className='flex sections' id='about'>
        <article id='profile-pic-container'>
            <img src={profilePic} alt="profile pic" id="profile-pic" />
        </article>
        <article id='about-text'>
            <h2>Edwin M. Escobar</h2>
            <h3>Full Stack Web Developer</h3>
            <ul>
                <li>I code code thanks to UT - Austin's Coding Bootcamp.</li>
                <li><a href="https://github.com/escowin" target="_blank" rel="noreferrer noopener">Github Repo</a></li>
                <li><a href="https://www.linkedin.com/in/escowin/" target="_blank" rel="noreferrer noopener">LinkedIn</a></li>
                <li><a href="mailto:edwin@escowinart.com">Email</a></li>
            </ul>
        </article>
      </section>
    );
  }
  
  export default About;