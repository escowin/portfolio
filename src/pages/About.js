import "../assets/css/about.css";
import profilePic from "../assets/images/profile-pic.jpg";
import Contact from "../components/Contact";
import portfolioData from "../assets/data";

function About() {
  const { name, bio } = portfolioData.info;
  return (
    <section className="section" id="about">
      <article id="profile-pic-wrapper">
        <img src={profilePic} id="profile-pic" alt="profile pic" />
      </article>
      <article id="profile-text">
        <h2>{name}</h2>
        <Contact />
      </article>

      <article id="bio">
        {bio.map((text, i) => (
          <p key={i}>{text}</p>
        ))}
      </article>
    </section>
  );
}

export default About;
