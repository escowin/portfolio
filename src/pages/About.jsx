import "../assets/css/about.css";
import profilePic from "../assets/img/profile-pic.jpg";
import Contact from "../components/Contact";
import portfolioData from "../assets/data";

function About() {
  const { name, bio } = portfolioData.info;
  return (
    <section className="section" id="about">
      <article id="profile-pic-wrapper">
        <img src={profilePic} id="profile-pic" alt="profile pic" className="rounded--full" />
      </article>
      <article id="profile-text">
        <h2 className="heading heading--secondary">{name}</h2>
        <Contact />
      </article>

      <article id="bio">
        {bio.map((text, i) => (
          <p key={i} className="text">{text}</p>
        ))}
      </article>
    </section>
  );
}

export default About;
