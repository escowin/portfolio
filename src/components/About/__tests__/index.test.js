import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import About from "..";
// if component is updated, snapshot becomes stale. write new snapshot, test runner cli u
describe('About component', () => {
    // render.about-test
    // - first-test | renders About component via JSX. (it(...) & test(...) are interchangeable)
    it('renders', () => {
        render(<About />);
    });

    // - second-test
    it('matches snapshot DOM node structure', () => {
        // render about
        const { asFragment } = render (<About />);
        expect(asFragment()).toMatchSnapshot();
    })
})

afterEach(cleanup);