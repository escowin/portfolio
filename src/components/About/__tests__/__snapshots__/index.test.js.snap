// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="section"
  >
    <h1
      id="about"
    >
      who am I?
    </h1>
    <img
      alt="cover"
      class="pic"
      src="cover.jpg"
      style="width: 100%;"
    />
  </section>
</DocumentFragment>
`;
